recipes/go4/run_go4_examples.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -eumx -o pipefail
+shopt -s failglob
+echo $GO4SYS
+echo $PREFIX
+
+pushd Go4ExampleSimple
+make clean
+make all -j$(nproc)
+go4analysis -random -number 100000
+popd
+
+pushd Go4ExampleUserSource
+make clean
+make all -j$(nproc)
```

Comment:
```suggestion
pushd Go4ExampleSimple
make clean
make CXX=${CXX} all -j$(nproc)
go4analysis -random -number 100000
popd

pushd Go4ExampleUserSource
make clean
make CXX=${CXX} all -j$(nproc)
```
Can you check if this allows you to remove `gxx` from the test dependencies on Linux?  In many cases the make file just needs to be "force-fed" `CXX` even though we explicitly define it.